### ISSUE TITLE
Wrong operation of Queue in ItemEntityMixin#tick

### ISSUE THREAD

**Issue by SettingDust** [MUYUTwilighter/saplanting_fabric#16]
1.20.1 fabric

https://mclo.gs/wNCxkO5
```
"Server thread" prio=4 Id=366 WAITING on java.util.concurrent.CompletableFuture$Signaller@417c7c29
	at java.base@17.0.8/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.CompletableFuture$Signaller@417c7c29
	at java.base@17.0.8/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1864)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3465)
	at java.base@17.0.8/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3436)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1898)
	at java.base@17.0.8/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2117)
	at net.minecraft.class_3215.getChunkOffThread(class_3215.java:3696)
	at net.minecraft.class_3215.method_12121(class_3215.java:3661)
	at net.minecraft.class_1937.method_8402(class_1937.java:204)
	at net.minecraft.class_1937.method_22350(class_1937.java:15049)
	at net.minecraft.class_1937.method_8500(class_1937.java:15044)
	at net.minecraft.class_1937.method_30092(class_1937.java:226)
	at net.minecraft.class_1937.method_8652(class_1937.java:213)
	at net.minecraft.class_1937.method_8501(class_1937.java:322)
	at cool.muyucloud.saplanting.util.PlantContext.plant(PlantContext.java:15)
	at net.minecraft.class_1542.handler$nab01c$saplanting$tick(class_1542.java:9102)
	at net.minecraft.class_1542.method_5773(class_1542.java:183)
	at net.minecraft.class_3218$$Lambda$25775/0x0000028a081021e8.accept(Unknown Source)
	at net.minecraft.class_3218.track(class_3218.java:28254)
	at net.minecraft.class_3218.redirect$mid000$observable$onTickNonPassenger(class_3218.java:28270)
	at net.minecraft.class_3218.method_18762(class_3218.java:739)
	at net.minecraft.class_3218$$Lambda$25772/0x0000028a08101980.accept(Unknown Source)
	at net.minecraft.class_1937.mixinextras$bridge$accept$219(class_1937.java)
	at net.minecraft.class_1937$$Lambda$25773/0x0000028a08101db0.call(Unknown Source)
	at com.bawnorton.neruina.handler.NeruinaTickHandler.safelyTickEntities$notTheCauseOfTickLag(NeruinaTickHandler.java:105)
	at net.minecraft.class_1937.wrapOperation$mfd000$neruina$catchTickingEntities(class_1937.java:18548)
	at net.minecraft.class_1937.method_18472(class_1937.java:480)
	at net.minecraft.class_3218.method_31420(class_3218.java:385)
	at net.minecraft.class_3218$$Lambda$25771/0x0000028a080ffda0.accept(Unknown Source)
	at net.minecraft.class_5574.method_31791(class_5574.java:54)
	at net.minecraft.class_3218.method_18765(class_3218.java:353)
	at net.minecraft.server.MinecraftServer.redirect$hfg000$carpet-fixes$replacedThisMethodIgnoreIt_CarpetFixesIsNotCausingLag(MinecraftServer.java:12828)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:897)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824)
	at net.minecraft.class_1132.method_3748(class_1132.java:105)
	at net.minecraft.server.MinecraftServer.handler$ghg000$carpet$modifiedRunLoop(MinecraftServer.java:21442)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:650)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265)
	at net.minecraft.server.MinecraftServer$$Lambda$24634/0x0000028a07f2dba0.run(Unknown Source)
	at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)
```

**Comment by MUYUTwilighter** [comment]
Can you provide a more detailed information about your situation (For example, the mod list)?
Saplanting 1.2.5 works fine on my machine (fabric 1.20.1 of course)

**Comment by SettingDust** [comment]
a bit harder to do that. it's my own modpack that contains too many mods.
maybe you can provide a version  log some verbose lines.
I'm holding the save

**Comment by MUYUTwilighter** [comment]
Issue causes were found and will be fixed in Saplanting 1.2.6.

Thank you for your post, the problem you faced would probably happen in other MC version of this mod.
